gems/aws-sdk-sqs/lib/aws-sdk-sqs/queue_poller.rb

```diff
@@ -419,9 +418,19 @@ def yield_messages(config, messages, stats, &block)
         end
       end
 
+      def dedupe_messages(messages)
+        u_messages = {}
+        messages.each do |msg|
```

Comment:
Not to play too much golf, but I think you could shorten this slightly by using `inject({})`.

Reply:
I think the recommendation is `each_with_object` and not inject (at least rubocop might say that..)

Reply:
I played around using `reverse.uniq.reverse!` on the messages. 
It unfortunately won't work since the duplicated messages with have different `receipt_handles` and the `uniq` will classifies these duplicates as separate messages. 

Going with `each_with_object`

Reply:
Scratch my above comment - it does work with block being passed in with .uniq - thanks Matt!